### ISSUE TITLE
Error: Could not authenticate via any gargle cred function

### ISSUE THREAD

**Issue by ogola89** [r-lib/gargle#151]
## What goes wrong

This is an issue I originally raised on the googleAuthR page and was directed here.

I am trying to use googleAuthR, which uses gargle, and I installed this package and successfully authenticated my own email while having a play with the package using:
`gar_auth()`

However, when I run a new session, it uses the cached email without any option to go ahead with it (which is fine). When instead, I run:
`gar_auth(email=another_email@domain.com)`

I get the following error:
`Error: Could not authenticate via any gargle cred function`

Traceback:
`Error: Could not authenticate via any gargle cred function
2.
stop("Could not authenticate via any gargle cred function", call. = FALSE)
1.
gar_auth(email = "another_email@domain.com")
`
Using different emails in the function call all produce the same error (including the one already cached and that works).

Using the cached email, though, doesn't throw any errors at all.

## Steps to reproduce the problem

```
library(googleAuthR)
library(googleAnalyticsR)
library(ggplot2)

gar_auth(email='another_email@domain.com')
```
This is the entire code up to this point.

## Expected output

Expect it to open google and authenticate as it did with the first email.

## Actual output

`Error: Could not authenticate via any gargle cred function`

## Session Info

R version 4.0.0 (2020-04-24)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19041)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.1252  LC_CTYPE=English_United Kingdom.1252    LC_MONETARY=English_United Kingdom.1252
[4] LC_NUMERIC=C                            LC_TIME=English_United Kingdom.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.2          googleAnalyticsR_0.8.0 googleAuthR_1.3.0     

loaded via a namespace (and not attached):
 [1] rstudioapi_0.11  knitr_1.29       magrittr_1.5     munsell_0.5.0    tidyselect_1.1.0 colorspace_1.4-1 R6_2.4.1        
 [8] rlang_0.4.7      fansi_0.4.1      httr_1.4.2       dplyr_1.0.1      tools_4.0.0      grid_4.0.0       gtable_0.3.0    
[15] xfun_0.15        cli_2.0.2        withr_2.2.0      askpass_1.1      ellipsis_0.3.1   openssl_1.4.2    assertthat_0.2.1
[22] digest_0.6.25    tibble_3.0.3     gargle_0.5.0     lifecycle_0.2.0  crayon_1.3.4     tidyr_1.1.1      purrr_0.3.4     
[29] vctrs_0.3.2      fs_1.5.0         curl_4.3         memoise_1.1.0    glue_1.4.1       compiler_4.0.0   pillar_1.4.6    
[36] scales_1.1.1     generics_0.0.2   jsonlite_1.7.0   pkgconfig_2.0.3

**Comment by jennybc** [comment]
I'm not sure how to help because this error is emitted by googleAnalyticsR and the similar code for packages I maintain that use gargle all work fine (I subbed out some email addresses after making the reprex, i.e. this code executes without error):

``` r
library(googledrive)

drive_auth(email = "jenny@example.com")
drive_user()
#> Logged in as:
#>   *  displayName: Jenny Bryan
#>   * emailAddress: jenny@example.com

library(googlesheets4)
#> 
#> Attaching package: 'googlesheets4'
#> The following objects are masked from 'package:googledrive':
#> 
#>     request_generate, request_make

gs4_auth(email = "jane.doe@gmail.com")
gs4_user()
#> [1] "jane.doe@gmail.com"
```

<sup>Created on 2020-08-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

So the problem seems to have something to do with googleAuthR and/or googleAnalyticsR.

**Comment by MarkEdmondson1234** [comment]
Is it expected behavior to authenticate the first time smoothly when you supply a non-cached email? Eg it goes through Oauth2 flow with an email pre-registered? This is something googleAuthR may get in the way of.

**Comment by jennybc** [comment]
Yeah, it should kick-off regular OAuth2 flow if there's no cache hit. And I just checked that this works for `googlesheets4::gs4_auth()`.

Plus @ogola89 reports:
> Using different emails in the function call all produce the same error (including the one already cached and that works).

**Comment by jennybc** [comment]
So it starts to sound like `gar_auth(email=)` isn't wired up correctly.

**Comment by MarkEdmondson1234** [comment]
Ok cool thanks, will sort that out

**Comment by Bertil** [comment]
I had the same problem. In my case it our Proxy that caused the Problem. I guess you get this error message when you cannot connect with the server
